www/js/splash/notifScheduler.ts

```diff
@@ -0,0 +1,269 @@
+};
+
+/* remove notif actions as they do not work, can restore post routing migration */
+// const setUpActions = () => {
+//     const action = {
+//         id: 'action',
+//         title: 'Change Time',
+//         launch: true
+//     };
+//     return new Promise((rs) => {
+//         cordova.plugins.notification.local.addActions('reminder-actions', [action], rs);
+//     });
+// }
+function debugGetScheduled(prefix) {
+  window['cordova'].plugins.notification.local.getScheduled((notifs) => {
+    if (!notifs?.length) return logDebug(`${prefix}, there are no scheduled notifications`);
+    const time = DateTime.fromMillis(notifs?.[0].trigger.at).toFormat('HH:mm');
+    //was in plugin, changed to scheduler
+    let scheduledNotifs = [];
+    scheduledNotifs = notifs.map((n) => {
+      const time = DateTime.fromMillis(n.trigger.at).toFormat('t');
+      const date = DateTime.fromMillis(n.trigger.at).toFormat('DDD');
+      return {
+        key: date,
+        val: time,
+      };
+    });
+    //have the list of scheduled show up in this log
+    logDebug(
+      `${prefix}, there are ${notifs.length} scheduled notifications at ${time} first is ${scheduledNotifs[0].key} at ${scheduledNotifs[0].val}`,
+    );
+  });
+}
+
+//new method to fetch notifications
+export const getScheduledNotifs = function () {
+  return new Promise((resolve, reject) => {
+    /* if the notifications are still in active scheduling it causes problems
+        anywhere from 0-n of the scheduled notifs are displayed 
+        if actively scheduling, wait for the scheduledPromise to resolve before fetching prevents such errors
+        */
+    if (isScheduling) {
+      console.log('requesting fetch while still actively scheduling, waiting on scheduledPromise');
+      scheduledPromise.then(() => {
+        getNotifs().then((notifs) => {
+          console.log('done scheduling notifs', notifs);
+          resolve(notifs);
+        });
+      });
+    } else {
+      getNotifs().then((notifs) => {
+        resolve(notifs);
+      });
+    }
+  });
+};
+
+//get scheduled notifications from cordova plugin and format them
+const getNotifs = function () {
+  return new Promise((resolve, reject) => {
+    window['cordova'].plugins.notification.local.getScheduled((notifs) => {
+      if (!notifs?.length) {
+        console.log('there are no notifications');
+        resolve([]); //if none, return empty array
+      }
+
+      const notifSubset = notifs.slice(0, 5); //prevent near-infinite listing
+      let scheduledNotifs = [];
+      scheduledNotifs = notifSubset.map((n) => {
+        const time = DateTime.fromMillis(n.trigger.at).toFormat('t');
+        const date = DateTime.fromMillis(n.trigger.at).toFormat('DDD');
+        return {
+          key: date,
+          val: time,
+        };
+      });
+      resolve(scheduledNotifs);
+    });
+  });
+};
+
+// schedules the notifications using the cordova plugin
+const scheduleNotifs = (scheme, notifTimes) => {
```

Comment:
You can also type `notifTimes` as an array of `DateTime`s.
If you had done this before, Typescript could have helped you catch the issue that caused the crash (because Typescript would have known `DateTime.toISO` returns a string, which does not have a property called `ts`.)
That's why type definitions are worth the effort!